src/libraries/System.Linq/src/System/Linq/AppendPrepend.SpeedOpt.cs

```diff
@@ -101,10 +101,10 @@ public override List<TSource> ToList()
                 if (count == 1)
                 {
                     // If GetCount returns 1, then _source is empty and only _item should be returned
-                    return new List<TSource>(1) { _item };
+                    return [_item];
```

Comment:
In the common case where the consumer doesn't add more to the returned list, this is increasing the memory allocated.

Reply:
Good catch, will revert these.

Reply:
Initial capacity shouldn't be an issue anymore as Roslyn will emit something like this now
```c#
List<TSource> list = new List<TSource>(1);
CollectionsMarshal.SetCount(list, 1);
Span<TSource> span = CollectionsMarshal.AsSpan(list);
span[0] = _item;
```
should we continue avoiding collection expressions in such cases?